app/src/main/java/com/codingblocks/cbonlineapp/mycourse/MyCourseRepository.kt

```diff
@@ -320,6 +322,8 @@ class MyCourseRepository(
         Clients.api.addToCart(id)
     }
 
+    suspend fun fetchLeaderboard(runId: String) = safeApiCall { Clients.api.leaderboardById(runId).awaitResponse() }
```

Comment:
there is still a small issue with this statement PTAL and try to understand what else can be done here